### ISSUE TITLE
With golang 1.20, govmomi will report issue validating certificate

### ISSUE THREAD

**Issue by lubronzhan** [kubernetes/cloud-provider-vsphere#736]
### What happened?

Opened an issue in govmomi as well
https://github.com/vmware/govmomi/issues/3174

Right now if i compile from master branch with golang 1.20, even with correct thumbprint config, govmomi will report error like 

```
E0710 05:47:19.253974       1 connectionmanager.go:147] Cannot connect to vCenter with err: Post "https://10.191.133.186:443/sdk": tls: failed to verify certificate: x509: cannot validate certificate for 10.191.133.186 because it doesn't contain any IP SANs
E0710 05:47:20.258996       1 connection.go:181] Failed to create new client. err: Post "https://10.191.133.186:443/sdk": tls: failed to verify certificate: x509: cannot validate certificate for 10.191.133.186 because it doesn't contain any IP SANs
E0710 05:47:20.259047       1 connection.go:67] Failed to create govmomi client. err: Post "https://10.191.133.186:443/sdk": tls: failed to verify certificate: x509: cannot validate certificate for 10.191.133.186 because it doesn't contain any IP SANs
E0710 05:47:20.259058       1 connectionmanager.go:147] Cannot connect to vCenter with err: Post "https://10.191.133.186:443/sdk": tls: failed to verify certificate: x509: cannot validate certificate for 10.191.133.186 because it doesn't contain any IP SANs
```

### What did you expect to happen?

CPI should be working without certificate validation error when using golang 1.20

### How can we reproduce it (as minimally and precisely as possible)?

1. Bumped Cloud-provider-vsphere to use golang1.20
2. Govmomi version is 0.30.4
3. Then compile and deploy Cloud-provider-vsphere with correct tlsthumbprint config
4. Error shows up

```
E0710 05:47:20.259058       1 connectionmanager.go:147] Cannot connect to vCenter with err: Post "https://10.191.133.186:443/sdk": tls: failed to verify certificate: x509: cannot validate certificate for 10.191.133.186 because it doesn't contain any IP SANs
```

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here
```

</details>

### Cloud provider or hardware configuration

<details>

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Kernel (e.g. `uname -a`)

<details>

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### Others

<details>

</details>

**Comment by lubronzhan** [comment]
Merged in https://github.com/vmware/govmomi/tree/v0.30.6. Will pick up this one for 1.27

**Comment by lubronzhan** [comment]
Fixed in 1.27.0
